secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/dcr/request/RegistrationRequestEntityValidatorFilter.java

```diff
@@ -0,0 +1,162 @@
+        Reject.ifNull(registrationRequestBuilder, "registrationRequestBuilder must be provided");
+        Reject.ifNull(jwtDecoder, "jwtDecoder must be provided");
+        Reject.ifNull(responseFactory, "responseFactory must be provided");
+        this.registrationEntitySupplier = registrationEntitySupplier;
+        this.acceptHeaderSupplier = acceptHeaderSupplier;
+        this.trustedDirectoryService = trustedDirectoryService;
+        this.registrationRequestBuilder = registrationRequestBuilder;
+        this.jwtDecoder = jwtDecoder;
+        this.responseFactory = responseFactory;
+    }
+
+    @Override
+    public Promise<Response, NeverThrowsException> filter(Context context, Request request, Handler next) {
+        final String transactionId = FAPIUtils.getFapiInteractionIdForDisplay(context);
+        log.debug("({}) Running RegistrationRequestEntityValidatorFilter", transactionId);
+        try {
+            String b64EncodedRegistrationRequestEntity = this.registrationEntitySupplier.apply(context, request);
+            RegistrationRequest registrationRequest = this.registrationRequestBuilder.build(transactionId,
+                    b64EncodedRegistrationRequestEntity);
+            context.asContext(AttributesContext.class).getAttributes().put(RegistrationRequest.REGISTRATION_REQUEST_KEY,
+                    registrationRequest);
+            log.info("({}) created context attribute " + RegistrationRequest.REGISTRATION_REQUEST_KEY, transactionId);
+            return next.handle(context, request);
+        } catch (DCRException exception){
+            List<String> acceptHeader = acceptHeaderSupplier.apply(context, request);
+            Response response = responseFactory.getResponse(transactionId, acceptHeader, Status.BAD_REQUEST,
+                    exception.getErrorFields());
+            log.info("({}) Failed to understand the Registration Request body: {}", transactionId,
+                    exception.getMessage(), exception);
+            return Promises.newResultPromise(response);
+        } catch (RuntimeException rte){
+            log.warn("({}) caught runtime exception while applying RegistrationRequestEntityValidatorFilter",
+                    transactionId, rte);
+            List<String> acceptHeader = acceptHeaderSupplier.apply(context, request);
+            Map<String, String> errorFields = new LinkedHashMap<>();
+            errorFields.put("error", "Server unable to process request");
+            errorFields.put("trace_id", transactionId);
+            Response response = responseFactory.getResponse(transactionId, acceptHeader,
+                    Status.INTERNAL_SERVER_ERROR, errorFields);
+            return Promises.newResultPromise(response);
+        }
+    }
+
+    /**
+     * Heaplet is used to get arguments from the IG config and create a RequestAndSsaSignatureValidationFilter
+     * on the IG heap.
+     */
+    public static class Heaplet extends GenericHeaplet {
```

Comment:
Document how to create the filter via config, see com.forgerock.sapi.gateway.mtls.TransportCertValidationFilter.Heaplet for an example